ecosystem/daos/jsondao.py

```diff
@@ -2,26 +2,103 @@
 from typing import Optional, List
 import os
 import json
+from pathlib import Path
+import shutil
+import toml
 
 from tinydb import TinyDB, Query
 
 from ecosystem.models import TestResult, StyleResult, CoverageResult, TestType
 from ecosystem.models.repository import Repository
 
 
+class EcosystemStorage:
+    """
+    Tell TinyDB how to read/write from files
+    (see https://tinydb.readthedocs.io/en/latest/extend.html)
+
+    File structure:
+
+    self.root
+    ├── members.json  # compiled file; don't edit manually
+    └── members
+        └── repo-name.toml
+
+    Database structure:
+
+    { tier: {  # e.g. Main, Community
+        index: repo  # `repo` is data from repo-name.toml
+    }}
+
+    """
+
+    def __init__(self, root_path):
+        self.toml_dir = Path(root_path, "members")
+
+    def _url_to_path(self, url):
+        repo_name = url.strip("/").split("/")[-1]
+        return self.toml_dir / f"{repo_name}.toml"
+
+    def read(self):
+        """
+        Search for TOML files and add to DB
+        """
+        if not self.toml_dir.is_dir():
+            # For TinyDB initialization
+            return None
+
+        data = {}
+        for path in self.toml_dir.glob("*"):
+            repo = toml.load(path)
+            tier = repo["tier"]
+            if tier not in data:
+                data[tier] = {}
+            index = len(data[tier].keys())
+            data[tier][index] = repo
+
+        return data
+
+    def write(self, data):
+        """
+        Write TOML files
+        """
+        # Erase existing TOML files
+        # (we erase everything to clean up any deleted repos)
+        if self.toml_dir.exists():
+            shutil.rmtree(self.toml_dir)
+
+        # Rewrite to human-readable TOML
```

Comment:
Nit, technically this is `write` rather than `rewrite` because you finished deleting everything already.